jung-algorithms/src/main/java/edu/uci/ics/jung/layout/model/LoadingCacheLayoutModel.java

```diff
@@ -17,47 +17,55 @@
   protected LoadingCache<N, P> locations =
       CacheBuilder.newBuilder().build(CacheLoader.from(() -> pointModel.newPoint(0, 0, 0)));
 
-  public LoadingCacheLayoutModel(LoadingCacheLayoutModel<N, P> other) {
-    super(other.graph, other.pointModel, other.width, other.height, other.depth);
-  }
+  public static class Builder<N, P> extends AbstractLayoutModel.Builder<N, P> {
```

Comment:
I prefer to create a builder via a static method (and then generally never actually deal with the Builder code directly at all):

    LoadingCacheLayoutModel layoutModel = LoadingCacheLayoutModel.builder()
        .graph(graph)
        .pointModel(pointModel)
        .build();

(I'm also not a big fan these days of the get/set method name prefixes where they don't add value, but that's a minor issue.)

If there's a parameter that's fundamental to the builder, you can set the builder creation method to take that parameter, e.g.:

        LoadingCacheLayoutModel layoutModel = LoadingCacheLayoutModel.builder(graph)
        .pointModel(pointModel)
        .build();

But that may not be appropriate here, since there are a few different such parameters.

Reply:
I like that. I moved the builder method in and out of the Builder class to try different ways. One problem may be that static build methods on both a parent and subclass may have the same type erasure and make the compiler complain. (one returns Parent.Builder and one returns Child.Builder). I need to look at more code examples.

Reply:
I forgot to add  a response to your get/set comment. I too, came from a C++ background and chafed years ago at the get/set/is java beans naming standard. Now, I am so used to it that  I cursed many times when I tried to call getNodes() instead of nodes() on your new Graph API. The power of writing a new library is that we can force everyone to adapt to what we like and away from long held (even when we disliked them at first) standards.

Reply:
Actually, I don't have a C++ background; most of my spurning of get()/set() comes from the direction that the Guava libraries have been moving (but it did take me a while to get used to it).

I'm not sure how to interpret your last comment; is that in favor of getting rid of get/set or an argument against doing so?  :)

Reply:
I'm just sayin' that you pay the price of having users grumble when they keep trying to call getNodes() because they are used to it, even if they never liked the javabeans style. I don't anticipate that I will have any influence on the design of anything in guava. I think that we sometimes want to weigh our style choices with the effect that it has on users.

Reply:
Just back from a morning run, I am amped up enough to tell you what i really think! (as if i don't always do that)
Let's take graph nodes() vs getNodes(). I've been programming to the javabeans specification since it came out in 1997. I happen to prefer nodes() over getNodes() but i recognize the javabeans specification intent to standardize. If I were writing a library and I put nodes() instead of getNodes(), I would be violating the https://en.wikipedia.org/wiki/Principle_of_least_astonishment
by putting my preference (my ego) ahead of user expectations. Many, many users will have a little 'stumble' when they try to call getNodes().